### PR TITLE
Use Combine for MessageGateway

### PR DESCRIPTION
**Pull request by Kharchevskyi**
This PR replace usage of Promises in `MessageGateway`

issue #290

----------------------------------

**Tests** _(delete all except exactly one)_:
- Difficult to test. Firstly need move to Combine
--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by tomholub** [comment]
Thank you - is this fully functional for me to be able to merge this work into master and potentially later make a release in this state?

**Comment by Kharchevskyi** [comment]
You can check as well, but as far as I tested, message was sent successfully, in both cases, when we compose a message and after creating a new backup

File: FlowCrypt/Controllers/Compose/ComposeViewController.swift [hunk 1]
```diff
@@ -216,61 +218,65 @@ extension ComposeViewController {
...
+                    guard let error = result.getError() else {
+                        return
+                    }
+
+                    self?.showAlert(error: error, message: "compose_error".localized)
+                },
+                receiveValue: { [weak self] in
+                    self?.handleSuccessfullySentMessage()
+                })
+            .store(in: &cancellable)
```
Comment (tomholub) [inline FlowCrypt/Controllers/Compose/ComposeViewController.swift:279]: I'll file an issue to create a slightly different convention here for increased clarity.

**Review by tomholub — APPROVED** [review]
Code looks great, with a note below. I'll build and test it now.

**Pull request merged**